core/internal/gethwrappers/abigen.go

```diff
@@ -134,15 +134,24 @@ func generateCode(fset *token.FileSet, fileNode *ast.File) []byte {
 }
 
 func getContractName(fileNode *ast.File) string {
-	// Grab the contract name. It's always the first type in the file.
```

Comment:
note this invariant isn't true for the VRFCoordinatorV2 for some reason, relying on XXABI should be more reliable